### PR TITLE
evaluate error condition codes

### LINKED ISSUE #681: Evaluate error codes in conditions
**What would you like to be added**:
Since gardener/gardener#2212 the `.status.conditions[]` have a new, optional `.codes[]` field, similar to `.status.lastError.codes[]`. It may contain a list of error codes that categorize certain problems/issue observed with the shoot cluster.

Please evaluate these error codes similar to how `.status.lastError.codes[]` are evaluated, and display them properly in the Shoot's condition chips (similar to the reconciliation icon that indicates user/infrastructure issues).

Also, please hide them by default from the "All projects" shoot list, similar to how shoots with `.status.lastError.codes[]` are hidden by default.

**Why is this needed**:
Make the life of Gardener administrators/operators easier by hiding problematic clusters by default if the operator cannot help making it healthy again.

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #681

**Special notes for your reviewer**:

**Release note**:

```improvement user
Evaluate `.status.conditions[].codes[]` field and show user errors on readiness tags
```

### CODE REVIEW

File: frontend/src/utils/errorCodes.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+'use strict'
+
+import isEmpty from 'lodash/isEmpty'
+import includes from 'lodash/includes'
+import every from 'lodash/every'
+import compact from 'lodash/compact'
+import uniq from 'lodash/uniq'
+import flatMap from 'lodash/flatMap'
+
+export function allErrorCodesFromLastErrorsOrConditions (lastErrorsOrConditionsarray) {
```
Comment (petersutter) [inline frontend/src/utils/errorCodes.js:26]: name suggestion:
`export function errorCodes(list) {`

File: frontend/src/components/StatusTag.vue [hunk 2]
```diff
@@ -141,6 +150,18 @@ export default {
...
+    },
+    allErrorCodeShortDescriptions () {
+      return map(this.condition.codes, code => get(errorCodes, `${code}.shortDescription`, code))
+    },
+    errorCodeShortDescriptionsText () {
+      return join(this.allErrorCodeShortDescriptions, ', ')
+    },
+    errorCodeDescriptions () {
+      return map(this.condition.codes, code => get(errorCodes, `${code}.description`, code))
+    },
```
Comment (petersutter) [inline frontend/src/components/StatusTag.vue:164]:
```js
get(errorCodes, [code, 'description'], code)
```

File: frontend/src/utils/errorCodes.js (hunk 1 above)
Comment (grolu) [reply frontend/src/utils/errorCodes.js]: done

File: frontend/src/components/StatusTag.vue (hunk 2 above)
Comment (grolu) [reply frontend/src/components/StatusTag.vue]: done

**Pull request merged**